Add node resource availability (#378)
eclipse-ankaios/ankaios: server/doc/swdesign/README.md

- impl
- utest

> #### Server receives agent node resource availability

Reviewer: This requirement does only document the call of the update. You need another one, that explains how the agent's resource availability is updated inside the ServerState. You can check how it was done for the other methods from ServerState and AgentMap.

`swdd~server-receives-resource-availability~1`

Status: approved

> When the Ankaios Server receives new a new agent load status, then the Ankaios Server shall trigger the ServerState to update its internal info regarding the connected agents' resource availability.

Reviewer: ```suggestion
When the Ankaios server receives a new agent load status, the Ankaios server shall trigger the ServerState to update the connected agent's resource availability with the provided new agent load status.
```
